internal/controller/correlation/middleware.go

```diff
@@ -32,18 +34,20 @@ func OnResponseComplete(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		begin := time.Now()
 		next.ServeHTTP(w, r)
-		correlationId := FromContext(r.Context())
-		if common.LoggingClient != nil {
-			common.LoggingClient.Trace("Response complete", clients.CorrelationHeader, correlationId, "duration", time.Since(begin).String())
+		correlationId := IdFromContext(r.Context())
+		lc := LoggingClientFromContext(r.Context())
+		if lc != nil {
+			LoggingClient.Trace("Response complete", clients.CorrelationHeader, correlationId, "duration", time.Since(begin).String())
```

Comment:
I would prefer to create a  local in line logger instance in the rare case `lc` is nil so we can remove the Global instance.

Reply:
sorry was in a hurry and didn't remove the  global instance, do you mean change to be like
```
if lc != nil {
    lc.Trace(...)
}
```
and remove the global `LoggingClient` instance ?

Reply:
Yes, that is what we need.